src/js/components/color-palette.react.js

```diff
@@ -0,0 +1,209 @@
+        const screenWidth = workspaceEl.clientWidth;
+
+        // Magic numbers
+        const COLOR_WIDTH = 22; // Width of each color div in the palette
+        const HORIZONTAL_POSITION_BUFFER = 250; // Horizontal distance to offset palette color picker
+        const VERTICAL_POSITION_BUFFER = 350; // Vertical distance to offset palette color picker
+
+        // An approximation of x based on size of widget and place within array
+        let x = (this.state.colors.length - i) * COLOR_WIDTH + HORIZONTAL_POSITION_BUFFER;
+        this.setState({ x: (screenWidth - x) });
+        this.setState({ y: (screenHeight - VERTICAL_POSITION_BUFFER) });
+
+        // Log the currentColor being edited + the position of the color within our internal color array
+        this.setState({ currentColor: color });
+        this.setState({ currentPosition: i });
+
+        this.setState({ displayPicker: !this.state.displayPicker });
+    }
+
+    /**
+     * Called to close the color picker from the color palette
+     */
+    onHide () {
+        this.setState({ displayPicker: !this.state.displayPicker });
+    }
+
+    /**
+     * Called when a user changes a color using the color picker. It affects the current color palette
+     *
+     * @param color - the new color
+     */
+    onChange (color) {
+        // Set the color picker to whatever new color the user has picked
+        let oldColor = this.state.currentColor;
+        let newColor = {
+            color: new Color(color.rgb),
+            count: 1
+        };
+        this.setState({ currentColor: newColor });
+
+        // Then update the current color array with the new color
+        let newColors = this.state.colors;
+        newColors[this.state.currentPosition] = newColor;
+        this.setState({ colors: newColors });
+
+        // Alert each individual widget to that a color has changed
+        // Each widget will have to check if the change applies to itself
+        EventEmitter.dispatch('color-palette:color-change', { old: oldColor.color, new: newColor.color });
+    }
+
+    /**
+     * Resets the color picker when a new scene has loaded
+     */
+    clearColors () {
+        this.setState({ colors: [] });
+    }
+
+    /**
+     * Official React lifecycle method
+     * Called every time state or props are changed
+     */
+    render () {
+        let colors = [];
+        if (this.state.colors) {
+            for (let i = 0; i < this.state.colors.length; i++) {
+                let color = this.state.colors[i];
+                let widgetStyle = { backgroundColor: color.color.getRgbaString() };
+
+                // This represents each squared div for a color in the color palette
+                colors.push(<div key={i} className='palette-color' onClick={ this.onClick.bind(null, color, i) }><div className='square' style={widgetStyle}></div></div>);
+            }
+        }
+
+        return (
+            <div>
+                <div className='colors'>
```

Comment:
Update class names to be namespaced (see above). (Including this one, although it doesn't have a style.)